aws-redshiftserverless-workgroup/src/main/java/software/amazon/redshiftserverless/workgroup/CreateHandler.java

```diff
@@ -34,73 +40,92 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-
-                logger.log(String.format("%s has successfully been created.", ResourceModel.TYPE_NAME));
-
-                return awsResponse;
-
-            } catch (ConflictException ex) {
-                if (retryCount >= MAX_RETRIES || !isRetriableWorkgroupException(ex)) {
-                    throw ex;
-                }
-
-                logger.log(String.format("Retrying CreateWorkgroup due to expected ConflictException: " +
-                        "%s. Attempt %d/%d", ex.getMessage(), retryCount + 1, MAX_RETRIES));
-                retryCount++;
-            }
-
-            try {
-                Thread.sleep(10000);
-            } catch (InterruptedException ie) {
-                Thread.currentThread().interrupt(); // Restore the interrupted status
-                throw new RuntimeException("Interrupted during retry wait", ie);
-            }
-        }
+    private boolean isWorkgroupStable(final Object awsRequest,
+                                      final RedshiftServerlessResponse awsResponse,
+                                      final ProxyClient<RedshiftServerlessClient> proxyClient,
+                                      final ResourceModel model,
+                                      final CallbackContext context) {
+
+        GetWorkgroupRequest getWorkgroupStatusRequest = GetWorkgroupRequest.builder()
+                .workgroupName(model.getWorkgroupName())
+                .build();
+
+        GetWorkgroupResponse getWorkgroupResponse = this.readWorkgroup(getWorkgroupStatusRequest, proxyClient);
+
+        logger.log(String.format("%s : Workgroup: %s has successfully been read.",
+                ResourceModel.TYPE_NAME, getWorkgroupResponse.workgroup().workgroupName()));
+
+        logger.log(getWorkgroupResponse.toString());
+
+        return getWorkgroupResponse.workgroup().status().equals(WorkgroupStatus.AVAILABLE);
+    }
+
+    private boolean isNamespaceStable(final Object awsRequest,
+                                      final RedshiftServerlessResponse awsResponse,
+                                      final ProxyClient<RedshiftServerlessClient> proxyClient,
+                                      final ResourceModel model,
+                                      final CallbackContext context) {
+
+        GetNamespaceRequest getNamespaceRequest = GetNamespaceRequest.builder()
+                .namespaceName(model.getNamespaceName())
+                .build();
+
+        GetNamespaceResponse getNamespaceResponse = this.readNamespace(getNamespaceRequest, proxyClient);
+
+        logger.log(String.format("%s : Namespace: %s has successfully been read.",
+                ResourceModel.TYPE_NAME, getNamespaceResponse.namespace().namespaceName()));
+
+        logger.log(getNamespaceResponse.toString());
+
+        return NAMESPACE_STATUS_AVAILABLE.equalsIgnoreCase(getNamespaceResponse.namespace().statusAsString());
```

Comment:
looks like we were checking workgroup using Enum, which seems safer to me, can we do something like that here, or why do we need equalsIgnoreCase instead?